src/backend/marsha/core/views.py

```diff
@@ -703,9 +704,19 @@ def get_context_data(self, **kwargs):
 
         passport, _ = LTIPassport.objects.get_or_create(playlist=playlist)
 
-        oauth_dict = {
-            "oauth_consumer_key": passport.oauth_consumer_key,
-        }
+        if settings.BYPASS_LTI_VERIFICATION:
+            oauth_dict = {
+                "oauth_consumer_key": passport.oauth_consumer_key,
+            }
+        else:
+            oauth_dict = {
+                "oauth_consumer_key": passport.oauth_consumer_key,
+                "oauth_timestamp": "",
+                "oauth_nonce": "",
+                "oauth_signature": "",
+                "oauth_version": "",
+                "oauth_signature_method": "",
```

Comment:
I don't understand this change.
It rings an alert : why do we need to modify the production code to run tests?

Reply:
We are in DevelopmentLTIView here, and I needed to add fields in the form to build a real LTI request.
This is not really production code, as this view is available only in debug.

Reply:
ok sorry! Is it normal that all the fields are empty then?

Reply:
Yes they just have to exist, as for our usecase, those fields are filled in test after an oauth signature, see https://github.com/openfun/marsha/pull/988/files#diff-81491be43853c68f3d629215c19b5b89a0cba6dfa7bac13638dc43f99aecad12R272-R299

Reply:
I find it convoluted and hard to understand why there must be those field empty, and it brings us back to the above debate about testing the LTI view. Maybe @lunika will see clearer and can advise what to do?

Reply:
As a real LTI request needs to post those parameters, we need them to launch a real request.
They were missing because this view were only used with BYPASS_LTI_VERIFICATION setting before.
This is not the case anymore with the tests in this PR.

Reply:
Yes it's a good idea, but while you're at it; why not fill the fields with the real values, (the way it's done in Richie or the OpenEdX Xblock), so that we can remove the bypass and make the dev view work closer to the real LMS scenario.

Reply:
Good idea too, but maybe I can do it in another PR, as it goes beyond this one.
I'm not sure this wouldn't have some side effects.